tests/test_yaml/stdlib_arrow_after_meta.yaml

```diff
@@ -0,0 +1,873 @@
+{% set version = "1.10.0" %}
+{% set cuda_enabled = cuda_compiler_version != "None" %}
+{% set build_ext_version = "5.0.0" %}
+{% set build_ext = "cuda" if cuda_enabled else "cpu" %}
+{% set proc_build_number = "0" %}
+{% set llvm_version = "15" %}
+
+# see https://github.com/apache/arrow/blob/apache-arrow-10.0.1/cpp/CMakeLists.txt#L88-L90
+{% set so_version = (version.split(".")[0] | int * 100 + version.split(".")[1] | int) ~ "." ~ version.split(".")[2] ~ ".0" %}
+
+package:
+  name: apache-arrow
+  version: {{ version }}
+
+source:
+  # fake source url to get version migrator to pass
+  url: https://github.com/scipy/scipy/archive/refs/tags/v{{ version }}.tar.gz
+  sha256: 3f9e587a96844a9b4ee7f998cfe4dc3964dc95c4ca94d7de6a77bffb99f873da
+  # # arrow has the unfortunate habit of changing tags of X.0.0 in the
+  # # lead-up until release -> don't use github sources on main
+  # # - url: https://github.com/apache/arrow/archive/refs/tags/apache-arrow-{{ version }}.tar.gz
+  # - url: https://www.apache.org/dyn/closer.lua/arrow/arrow-{{ version }}/apache-arrow-{{ version }}.tar.gz?action=download
+  #   fn: apache-arrow-{{ version }}.tar.gz
+  #   sha256: 01dd3f70e85d9b5b933ec92c0db8a4ef504a5105f78d2d8622e84279fb45c25d
+  #   patches:
+  #     # workaround for https://github.com/apache/arrow/issues/37692
+  #     - patches/0001-fixture-teardown-should-not-fail-test.patch
+  # testing-submodule not part of release tarball
+  # - git_url: https://github.com/apache/arrow-testing.git
+  #   git_rev: ad82a736c170e97b7c8c035ebd8a801c17eec170
+  #   folder: testing
+
+build:
+  number: 0
+  # for cuda support, building with one version is enough to be compatible with
+  # all later versions, since arrow is only using libcuda, and not libcudart.
+  skip: true  # [cuda_compiler_version not in ("None", cuda_compiler_version_min)]
+  # arrow promises API- & ABI-compatibility along SemVer, see #1096
+
+requirements:
+  build:
+    - {{ compiler("c") }}
+    - {{ compiler("cxx") }}
+    - {{ compiler("cuda") }}                 # [cuda_compiler_version != "None"]
+    # needs to run protoc & grpc_cpp_plugin
+    - libgrpc                                # [build_platform != target_platform]
+    - libprotobuf                            # [build_platform != target_platform]
+    # needed for gandiva
+    - clangdev {{ llvm_version }}            # [build_platform != target_platform]
+    - llvmdev {{ llvm_version }}             # [build_platform != target_platform]
+    - gnuconfig                              # [build_platform != target_platform]
+    - cmake
+    - ninja
+    # necessary for vendored jemalloc
+    - autoconf     # [linux]
+    - make         # [linux]
+  host:
+    - {{ stdlib("c") }}
```

Comment:
This should be in `build`.

Reply:
Then I think I misunderstood the intention of `{{ stdlib }}`. Conceptually, if we compile against something and need its symbols at runtime, shouldn't that be a host-dependency (regardless of whether we actually ship these bits ourselves)? 🤔

Reply:
`stdlib` is going to be `sysroot_linux-64` for linux and the sysroot should be in build.

Reply:
I guess it's a distinction without a substantial difference as long as we have a strong run-export on the stdlib. But since we have a choice how to introduce this, why not do the philosophically "cleaner" appoach?

Reply:
> I guess it's a distinction without a substantial difference as long as we have a strong run-export on the stdlib. 

We don't want the run-export on the stdlib. That's the whole point.

For eg in your case
```
build:
  - {{ compiler('c') }}
host:
  - {{ stdlib('c') }}
```
gets rendered to 
```
build:
  - gcc_linux-64 12.*
host:
  - sysroot_linux-64 2.17
```
Then it's ambiguous what sysroot version is used in build.

What we want is
```
build:
  - gcc_linux-64 12.*
  - sysroot_linux-64 2.17
host:
```
which is the rendered form of
```
build:
  - {{ compiler('c') }}
  - {{ stdlib('c') }}
host:
```

Reply:
> We don't want the run-export on the stdlib.

But that's what we [have](https://github.com/conda-forge/linux-sysroot-feedstock/blob/6b2306dea0e904367f32dcd37a72f5d0cbfdd92b/recipe/meta.yaml#L154-L156) right now? Where do we get the run-time requirement of the produced package from?

> Then it's ambiguous what sysroot version is used in build.

Yeah, I realize now that we cannot use an arbitrary sysroot in host, because we're bound to what the actual build environment has available in the image; IOW, we cannot get around the fact that we don't control/ship the c-stdlib.

Reply:
> But that's what we [have](https://github.com/conda-forge/linux-sysroot-feedstock/blob/6b2306dea0e904367f32dcd37a72f5d0cbfdd92b/recipe/meta.yaml#L154-L156) right now? Where do we get the run-time requirement of the produced package from?

Sorry. I thought you meant a run_export of the stdlib in the gcc package.